spec/system/new_topic_list_spec.rb

```diff
@@ -0,0 +1,444 @@
+          new_topic_in_category,
+          new_reply_with_tag,
+          new_topic_with_tag,
+        ].each { |topic| expect(topic_list).to have_topic(topic) }
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+      end
+
+      it "the All tab is the default is the default tab" do
+        visit("/new")
+
+        expect(tabs_toggle.all_tab).to be_active
+        expect(tabs_toggle.replies_tab).to be_inactive
+        expect(tabs_toggle.topics_tab).to be_inactive
+      end
+
+      it "respects the s (scope) query param and activates the appropriate tab" do
+        visit("/new?s=topics")
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_inactive
+        expect(tabs_toggle.topics_tab).to be_active
+
+        visit("/new?s=replies")
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_active
+        expect(tabs_toggle.topics_tab).to be_inactive
+      end
+
+      it "shows only new topics when the user switches to the Topics tab" do
+        visit("/new")
+        tabs_toggle.topics_tab.click
+
+        expect(topic_list).to have_topics(count: 3)
+        [new_topic, new_topic_in_category, new_topic_with_tag].each do |topic|
+          expect(topic_list).to have_topic(topic)
+        end
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_inactive
+        expect(tabs_toggle.topics_tab).to be_active
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+
+        expect(current_url).to end_with("/new?s=topics")
+      end
+
+      it "shows only topics with new replies when the user switches to the Replies tab" do
+        visit("/new")
+        tabs_toggle.replies_tab.click
+
+        expect(topic_list).to have_topics(count: 3)
+        [new_reply, new_reply_in_category, new_reply_with_tag].each do |topic|
+          expect(topic_list).to have_topic(topic)
+        end
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_active
+        expect(tabs_toggle.topics_tab).to be_inactive
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+
+        expect(current_url).to end_with("/new?s=replies")
+      end
+
+      it "strips out the s (scope) query params when switching back to the All tab from any of the other tabs" do
+        visit("/new")
+        tabs_toggle.replies_tab.click
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_active
+        try_until_success { expect(current_url).to end_with("/new?s=replies") }
```

Comment:
Is there a reason we need `try_until_success` only in this test case?

Reply:
Not really sure, I looked for existing tests that had assertions for current URL and found only one and it wrapped the assertion with `try_until_success`:

https://github.com/discourse/discourse/blob/f8cd1da92a7dd093b8c97983f7c1e6f00c9147b2/spec/system/topic_page_spec.rb#L31-L33

I tried to remove it, but the test became flakey.

Reply:
Do we know the reason as to why the test became flaky?

Reply:
I think there's a delay from the time the user clicks on the tab/button that changes the URL to the time Ember updates the URL in the browser, and this delay can sometimes be large enough to make the `current_url` assertion fail. That said, I tried removing the `try_until_success` block again to dig into the problem, but the test passed consistently several times and couldn't get it to fail even once, so I've gone ahead and removed the `try_until_success` block for now.